tests/drivers/clock_control/stm32_clock_configuration/stm32n6_devices/boards/devices_on_icx.overlay

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2025 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* Core and device clock set up
+ *
+ * Warning: This overlay performs configuration from clean sheet.
+ * It is assumed that it is applied after clear_clocks.overlay file.
+ */
+
+&clk_hsi {
+	status = "okay";
+	clock-frequency = <DT_FREQ_M(64)>;
+	hsi-div = <1>;
+};
+
+&cpusw {
+	status = "okay";
+	clocks = <&rcc STM32_SRC_IC1 CPU_SEL(3)>;
+	clock-frequency = <DT_FREQ_M(600)>;
+};
+
+&ic1 {
+	status = "okay";
+	pll-src = <1>;
+	ic-div = <2>;
+};
+
+&ic2 {
+	status = "okay";
+	pll-src = <1>;
+	ic-div = <3>;
+};
+
+&ic6 {
+	status = "okay";
+	pll-src = <1>;
+	ic-div = <2>;
+};
+
+&ic7 {
+	status = "okay";
+	pll-src = <1>;
+	ic-div = <2>;
+};
+
+&ic8 {
+	status = "okay";
+	pll-src = <1>;
+	ic-div = <2>;
+};
+
+&ic9 {
+	status = "okay";
+	pll-src = <1>;
+	ic-div = <2>;
```

Comment:
In order to test more cases, you could use different PLLs for IC8, 9 and 10 (source clocks of the peripherals), as well as different dividers

Reply:
Makes sense, i'll look at that.